tests/lib/rules/no-constant-condition.js

```diff
@@ -358,18 +371,6 @@ ruleTester.run("no-constant-condition", rule, {
             code: "if(''+[]) {}",
             errors: [{ messageId: "unexpected", type: "BinaryExpression" }]
         },
-        {
-            code: "if([a]==[a]) {}",
-            errors: [{ messageId: "unexpected", type: "BinaryExpression" }]
-        },
-        {
-            code: "if([a] - '') {}",
```

Comment:
This was a false positive. If `a = 1` the condition is true, if `a =0` it is false.